File: code/algo/getoffer/go/3/uniquearr_subset.go
package main

func subsets(nums []int) (ans [][]int) {
	n := len(nums)
	for mask := 0; mask < 1<<n; mask++ {
		set := []int{}
		for i, v := range nums {
			if mask>>i&1 > 0 {
				set = append(set, v)
			}
		}
		ans = append(ans, append([]int(nil), set...))
	}
	return
}

File: code/algo/getoffer/c++/3/uniquearr_subset.cpp
#include <vector>

using namespace std;

vector<vector<int>> subsets(vector<int> &nums) {
  vector<vector<int>> ans;
  int n = nums.size();
  for (int mask = 0; mask < (1 << n); ++mask) {
    vector<int> set;
    for (int i = 0; i < n; ++i) {
      if (mask & (1 << i)) {
        set.push_back(nums[i]);
      }
    }
    ans.push_back(set);
  }
  return ans;
}
